addons/web/static/tests/views/list_tests.js

```diff
@@ -2980,20 +2980,119 @@ QUnit.module('Views', {
+                    if (moves == 3) assert.deepEqual(args, {"model": "foo", "ids": [1, 4, 2, 3], "field": "int_field"});
+                    moves += 1;
+
+                //We simulate what happens in the python resequence method
+                //this is to make sure that the order we get from the server is coherent with the order in the frontend.
+                //if this is the case, this means that refreshing the page would magically reorder records.
+                for (var i in args.ids) {
+                    var offset = args.offset ? Number(args.offset) : 0;
+                    this.data[args.model].records[args.ids[i]-1].int_field = Number(i) + offset;
+                }
+                    return $.when();
+                }
+                return this._super.apply(this, arguments);
+            },
+        });
+
+        assert.strictEqual(list.$('tbody tr:eq(0) td:last').text(), '1',
+            "default should be sorted by id");
+        assert.strictEqual(list.$('tbody tr:eq(1) td:last').text(), '2',
+            "default should be sorted by id");
+        assert.strictEqual(list.$('tbody tr:eq(2) td:last').text(), '3',
+            "default should be sorted by id");
+        assert.strictEqual(list.$('tbody tr:eq(3) td:last').text(), '4',
+            "default should be sorted by id");
+
+        testUtils.dragAndDrop(
+            list.$('.ui-sortable-handle').eq(3),
+            list.$('tbody tr').eq(2),
+            {position: 'top'}
+        );
+
+        assert.deepEqual(_.pluck(this.data.foo.records, 'int_field'), [0, 1, 3, 2],
+            "the int_field (sequence) should have been correctly updated");
+
+        testUtils.dragAndDrop(
+            list.$('.ui-sortable-handle').eq(2),
+            list.$('tbody tr').eq(1),
+            {position: 'top'}
+        );
+
+        assert.deepEqual(_.pluck(this.data.foo.records, 'int_field'), [0, 2, 3, 1],
+            "the int_field (sequence) should have been correctly updated");
+
+        testUtils.dragAndDrop(
+            list.$('.ui-sortable-handle').eq(1),
+            list.$('tbody tr').eq(3),
+            {position: 'top'}
+        );
+
+        assert.deepEqual(_.pluck(this.data.foo.records, 'int_field'), [0, 1, 3, 2],
+            "the int_field (sequence) should have been correctly updated");
+
+        testUtils.dragAndDrop(
+            list.$('.ui-sortable-handle').eq(2),
+            list.$('tbody tr').eq(1),
+            {position: 'top'}
+        );
+
+        assert.deepEqual(_.pluck(this.data.foo.records, 'int_field'), [0, 2, 3, 1],
+            "the int_field (sequence) should have been correctly updated");
+
+        list.destroy();
+    });
+
     QUnit.test('editable list with handle widget', function (assert) {
         assert.expect(12);
 
+        // resequence makes sense on a sequence field, not on arbitrary fields
```

Comment:
Not really meaningless.
But they did check that the data was ordered in the js in a way that was different from the order given by int_field. Since we reorder the data to be coherent with the int_field value, these tests broke.